src/compas/geometry/primitives/polyline.py

```diff
@@ -258,11 +259,134 @@ def transform(self, T):
             self.points[index].y = point[1]
             self.points[index].z = point[2]
 
+    def split_at_corners(self, angle_threshold):
+        """Splits a polyline at corners larger than the given angle_threshold
 
+        Parameters:
+        -----------
+        angle_threshold: radians
+
+        Returns
+        -------
+        list of polylines
+
+        """
+
+        corner_ids = []
+        split_polylines = []
+        points = self.points
+        seg_ids = list(range(len(self.lines)))
+
+        if self.is_closed():
+            seg_ids.append(0)
+
+        for seg1, seg2 in pairwise(seg_ids):
+            angle = self.lines[seg1].vector.angle(self.lines[seg2].vector)
+            if angle >= angle_threshold:
+                corner_ids.append(seg1+1)
+
+        if self.is_closed() and len(corner_ids) > 0:
+            if corner_ids[-1] != len(points):
+                corner_ids = [corner_ids[-1]] + corner_ids
+        else:
+            corner_ids = [0] + corner_ids + [len(points)]
+
+        for id1, id2 in pairwise(corner_ids):
+            if id1 < id2:
+                split_polylines.append(Polyline(points[id1:id2+1]))
```

Comment:
I have no strong opinion about this. I definitely understand the reason to copy these points to avoid any mishaps later on in the function, but as it seems to work I would just leave it as is. 

Performance-wise I thought this would be the most efficient ( but I'm definitely no expert 😂 )

If you feel that it would improve the code by copying feel free to add.

Reply:
I am ambivalent about this, weighing performance against hard to spot bugs. So, I'd say leave it as it is until someone complains ;)